### DISCUSSION TITLE
Running Windows 2019 server produces an exit code of 1

### DISCUSSION THREAD

**Discussion by peter-ro** [Vampire/setup-wsl#11] [Q&A] [answered]
When integrating the following into my repo:

- uses: Vampire/setup-wsl@v1
  with:
      cache: 'false'
      distribution: Ubuntu-18.04
      set-as-default: 'true'

Returns an exit code of 1 but the installation of file and directories looks like it ran successfully - unsure if the distribution is installed on the system. We are running this within an Azure cloud environment on a Windows 2019 Server - which is a self-hosted github runner.

Is it because the following command needs to be ran? 
wsl-shell-command: bash -c "sudo -u test bash --noprofile --norc -euo pipefail "

**Comment by peter-ro** [comment]
Please see above comment: https://github.com/Vampire/setup-wsl/issues/10#issue-951114668

**Comment by peter-ro** [comment]
The exit code 1 occurs when it tries to install Ubuntu-18.04

**Comment by Vampire** [comment]
Besides that `cache` is not a valid input to my action, but `use-cache` would be, what is the output of the action?
Can you provide the log files?
![grafik](https://user-images.githubusercontent.com/325196/126788095-17117b98-590b-44ee-943d-86fc545ba671.png)
Maybe also set a repository secret with name `ACTIONS_STEP_DEBUG` and value `true` first and rerun the job to get more verbose logging.

**Comment by peter-ro** [comment]
Apologies, I did not mean cache but use-cache. The output of the job is:

Run actions/setup-wsl@v1.1.0
Verify Windows Environment
Install Distribution
Error: The process 'C:\actions-runner\_work\_tool\Ubuntu\16.4.0\x64\ubuntu1604.exe' failed with exit code 1

Attached are the logs.

[GA-WSL-01-logs.zip](https://github.com/Vampire/setup-wsl/files/6869327/GA-WSL-01-logs.zip)

**Comment by peter-ro** [comment]
Here is a more detailed look at the logs with `ACTIONS_STEP_DEBUG `set to true.

[GA-WSL-02-logs.zip](https://github.com/Vampire/setup-wsl/files/6869509/GA-WSL-02-logs.zip)

**Comment by Vampire** [comment, marked as answer]
I moved this to a discussion, as I don't think this is an issue with my action.
I tried the very same options on the GHA hosted windows server 2019 environment and there it worked flawlessly.
So I can just assume that something on your self-hosted runner is not setup properly.
Did you maybe not enable WSL on it? That's nothing my action can do as it requires administrative privileges and a restart.
The logs you attached show this:
```2021-07-23T14:24:00.6529232Z [command]C:\actions-runner\_work\_tool\Ubuntu\16.4.0\x64\ubuntu1604.exe install --root
2021-07-23T14:24:00.7669754Z Installing, this may take a few minutes...
2021-07-23T14:24:00.7685570Z 
2021-07-23T14:24:00.7728886Z WslRegisterDistribution failed with error: 0x80070005
2021-07-23T14:24:00.7736104Z 
2021-07-23T14:24:00.7744752Z Error: 0x80070005 Access is denied.
2021-07-23T14:24:00.7745257Z 
2021-07-23T14:24:00.7745579Z 
2021-07-23T14:24:00.7745842Z 
2021-07-23T14:24:00.7746705Z ::endgroup::
2021-07-23T14:24:00.7747013Z ##[endgroup]
2021-07-23T14:24:01.4524429Z ##[debug]Error: The process 'C:\actions-runner\_work\_tool\Ubuntu\16.4.0\x64\ubuntu1604.exe' failed with exit code 1
```

So you should probably Google for how to resolve error 0x80070005 when installing ubuntu wsl distribution and try to fix that.
